### PR TITLE
Update mimir-prometheus, add support for `align_evaluation_time_on_interval` for rule groups.

### PR DESCRIPTION
**Pull request by pstibrany**
#### What this PR does
This PR updates mimir-prometheus and adds support for `align_evaluation_time_on_interval` field in rule groups.

#### Checklist

- [x] Tests updated
- [na] Documentation added
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

**Review by replay — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -34,6 +34,7 @@
 * [ENHANCEMENT] Update Docker base images from `alpine:3.16.2` to `alpine:3.17.1`. #3898
 * [ENHANCEMENT] Ingester: Add new `/ingester/tsdb_metrics` endpoint to return tenant-specific TSDB metrics. #3923
 * [ENHANCEMENT] Query-frontend: CLI flag `-query-frontend.max-total-query-length` and its associated YAML configuration is now stable. #3882
+* [ENHANCEMENT] Ruler: rule groups now support optional and experimental `align_evaluation_time_on_interval` field, which causes all evaluations to happen on interval-aligned timestamp.
```
Comment (pracucci) [inline CHANGELOG.md:37]:
```suggestion
* [ENHANCEMENT] Ruler: rule groups now support optional and experimental `align_evaluation_time_on_interval` field, which causes all evaluations to happen on interval-aligned timestamp. #4013
```

File: docs/sources/mimir/operators-guide/configure/about-versioning.md [hunk 2]
```diff
@@ -51,6 +51,7 @@ The following features are currently experimental:
   - Disable alerting and recording rules evaluation on a per-tenant basis
     - `-ruler.recording-rules-evaluation-enabled`
     - `-ruler.alerting-rules-evaluation-enabled`
+  - Aligning of evaluation timestamp on interval ()
```
Comment (pracucci) [inline docs/sources/mimir/operators-guide/configure/about-versioning.md:54]:
```suggestion
  - Aligning of evaluation timestamp on interval (`align_evaluation_time_on_interval`)
```

**Review by pracucci — APPROVED** [review]
LGTM (modulo a couple of nits)

File: CHANGELOG.md (hunk 1 above)
Comment (pstibrany) [reply CHANGELOG.md:37]: Fixed in https://github.com/grafana/mimir/pull/4014.

File: docs/sources/mimir/operators-guide/configure/about-versioning.md (hunk 2 above)
Comment (pstibrany) [reply docs/sources/mimir/operators-guide/configure/about-versioning.md:54]: Fixed in https://github.com/grafana/mimir/pull/4014.

**Pull request merged**